### PR TITLE
 tsdb/cloudwatch: always return underlying metrics query errors

### PR DESCRIPTION
**Pull request by loshz**
**What this PR does / why we need it**:
Some of the methods used to aggregate metrics from CloudWatch do not always include the underlying error. This makes debugging issues related to these metrics a little difficult.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
N/A

### CODE REVIEW

File: pkg/tsdb/cloudwatch/metric_find_query.go [hunk 1]
```diff
@@ -699,7 +699,7 @@ func (e *CloudWatchExecutor) ec2DescribeInstances(region string, filters []*ec2.
 			return !lastPage
 		})
 	if err != nil {
-		return nil, errors.New("Failed to call ec2:DescribeInstances")
+		return nil, fmt.Errorf("Failed to call ec2:DescribeInstances, %v", err)
```
Comment (kylebrandt) [inline pkg/tsdb/cloudwatch/metric_find_query.go:702]: @marefr / @bergquist  Should this be `%w`, or maybe use errutil?

Comment (syscll) [reply pkg/tsdb/cloudwatch/metric_find_query.go:702]: Happy to change to using `%w`, although other functions in this file don’t currently do so. I can also update them if you like?

Comment (bergquist) [reply pkg/tsdb/cloudwatch/metric_find_query.go:702]: Personally I prefer `errutil.Wrap` https://github.com/grafana/grafana/blob/master/pkg/util/errutil/errors.go#L13 but either works :)

@syscll updating all cases in this file would be great.

**Comment by aknuds1** [comment]
Great initiative!

File: pkg/tsdb/cloudwatch/metric_find_query.go (hunk 1 above)
Comment (syscll) [reply pkg/tsdb/cloudwatch/metric_find_query.go]: I decided to stick with the stdlib as to remove the need for another package import. The `xerrors` pkg wrap functionality seems to be extremely similar to the new go1.13 wrapping feature - hope that's ok!

**Comment by syscll** [comment]
PTAL

**Review by aknuds1 — APPROVED** [review]

**Review by kylebrandt — APPROVED** [review]
Thank you!

**Comment by aknuds1** [comment]
I'm going to close and reopen to see if that triggers the CircleCI check, as it's currently stuck.

**Comment by aknuds1** [comment]
Thank you for helping make Grafana better!

**Comment by syscll** [comment]
No problem - thanks for making awesome software!

**Pull request merged**